File: app/src/main/java/com/elishaazaria/sayboard/ime/recognizers/providers/RecognizerSourceProvider.java
package com.elishaazaria.sayboard.ime.recognizers.providers;

import com.elishaazaria.sayboard.ime.recognizers.RecognizerSource;

import java.util.List;

public interface RecognizerSourceProvider {
    void loadSources(List<RecognizerSource> recognizerSources);
}

File: app/src/main/java/com/elishaazaria/sayboard/ime/recognizers/providers/RecognizerSourceProvider.kt
package com.elishaazaria.sayboard.ime.recognizers.providers

import com.elishaazaria.sayboard.data.InstalledModelReference
import com.elishaazaria.sayboard.ime.recognizers.RecognizerSource

interface RecognizerSourceProvider {
    fun getInstalledModels(): Collection<InstalledModelReference>

    fun recognizerSourceForModel(localModel: InstalledModelReference): RecognizerSource?
}